shaders/terrain_tessellation/slang/terrain.frag.slang

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2025, Sascha Willems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+[[vk::binding(1, 0)]] Sampler2D samplerHeight;
+[[vk::binding(2, 0)]] Sampler2DArray samplerLayers;
+
+struct DSOutput
+{
+	float4 Pos : SV_POSITION;
+	float3 Normal;
+	float2 UV;
+	float3 ViewVec;
+	float3 LightVec;
+	float3 EyePos;
+	float3 WorldPos;
+};
+
+float3 sampleTerrainLayer(float2 inUV)
+{
+	// Define some layer ranges for sampling depending on terrain height
+	float2 layers[6];
+	layers[0] = float2(-10.0, 10.0);
+	layers[1] = float2(5.0, 45.0);
+	layers[2] = float2(45.0, 80.0);
+	layers[3] = float2(75.0, 100.0);
+	layers[4] = float2(95.0, 140.0);
+	layers[5] = float2(140.0, 190.0);
```

Comment:
The glsl shader uses different values for layers[4] and layers[5]. Should probably be unified.